classes/publication/Repository.php

```diff
@@ -394,7 +394,7 @@ public function edit(Publication $publication, array $params): Publication
                 $submissionContext = Services::get('context')->get($submission->getData('contextId'));
             }
 
-            $supportedLocales = $submissionContext->getSupportedSubmissionLocales();
+            $supportedLocales = $publication->getLanguages($submissionContext->getSupportedSubmissionMetadataLocales());
```

Comment:
After thinking about it again: I think we would need submission->getLanguages here - - maybe also to be on a safe side...